utils/das_data_sorter.py

```diff
@@ -1,76 +1,179 @@
-data = pd.read_csv(args.file)
-
-milliseconds = data["time"]
-gps_course = data["gps_course"]
-gps_speed = data["gps_speed"]
-tempC = data["thermoC"]
-tempF = data["thermoF"]
-cadence = data["cadence"]
-power = data["power"]
-reed_vel = data["reed_velocity"]
-reed_dis = data["reed_distance"]
-
-time = []
-if args.unit == "seconds" or args.unit == "s":
-    time = milliseconds/1000
-elif args.unit == "minutes" or args.unit =="m":
-    time = milliseconds/1000/60
-
-# sorts time into specific time unit
-new_time = []
-previous_time = 0
-for i in range(len(time)):
-    if time[i] > previous_time:
-        # sets new previous_time, pushes current millisecond_indices into seconds array, creates new millisecond_indices
-        # millisecond_indices is an array of indexes for milliseconds within the same specified unit
-        if previous_time != 0:
-            # ignores first iteration
-            new_time.append(millisecond_indices)
-
-        previous_time = ceil(time[i])
-        millisecond_indices = []
-
-    # pushes index into current millisecond_indices
-    millisecond_indices.append(i)
-
-new_time.append(millisecond_indices)
-
-# writes newly-manipulated data into new csv file
-final = pd.DataFrame({
-    "time": range(1, len(new_time)+1),
-    "gps_course": avg_Data(gps_course),
-    "gps_speed": avg_Data(gps_speed),
-    "tempC": avg_Data(tempC),
-    "tempF": avg_Data(tempF),
-    "cadence": avg_Data(cadence),
-    "power": avg_Data(power),
-    "reed_velocity": avg_Data(reed_vel),
-    "reed_distance": avg_Data(reed_dis)
-})
-final.to_csv(args.output, index=False)
+class DasSort:
+    def __init__(self, file_input:pd.DataFrame, file_output:str, unit:str) -> None:
+        self.indexes = None
+        self.time = self.convert_time(file_input["time"], unit)
+        self.gps = self.average_data(file_input["gps"])
+        self.gps_location = self.location_data(file_input["gps_location"])
+        self.gps_course = self.average_data(file_input["gps_course"])
+        self.gps_speed = self.average_data(file_input["gps_speed"])
+        self.gps_satellites = self.average_data(file_input["gps_satellites"])
+        self.ax = self.average_data(file_input["aX"])
+        self.ay = self.average_data(file_input["aY"])
+        self.az = self.average_data(file_input["aZ"])
+        self.gx = self.average_data(file_input["gX"])
+        self.gy = self.average_data(file_input["gY"])
+        self.gz = self.average_data(file_input["gZ"])
+        self.thermoc = self.average_data(file_input["thermoC"])
+        self.thermof = self.average_data(file_input["thermoF"])
+        self.pot = self.average_data(file_input["pot"])
+        self.cadence = self.average_data(file_input["cadence"])
+        self.power = self.average_data(file_input["power"])
+        self.reed_velocity = self.average_data(file_input["reed_velocity"])
+        self.reed_distance = self.average_data(file_input["reed_distance"])
+
+        self.write_to_output_file(file_output)
```

Comment:
This should not be ran inside the class instructor. This should be ran by the class instance eg. `das_sort.write_to_output_file(args.output)`. If you do this, you also don't need to include the `file_output` argument.